### ISSUE TITLE
EventHub plugin reports strange values periodically

### ISSUE THREAD

**Issue by ilya-scale** [kedacore/keda#4250]
### Report

I have connected KEDA to event hub

```yml
      spec:
        minReplicaCount: 2
        maxReplicaCount: 8
      triggers:
      - type: azure-eventhub
        metadata:
          storageAccountName: myaccount
          blobContainer: checkpoints
          eventHubNamespace: mynamespace
          eventHubName: myhub
          consumerGroup: myconsumer
          checkpointStrategy: blobMetadata
          unprocessedEventThreshold: "5000"
```

There seem to be a number of issues with the connection:

1. The connection seems to work but I often get very strange values. I monitor them via grafana and prometheus monitor.
The value is negative and extremely low:

![image](https://user-images.githubusercontent.com/105273784/219648250-e2d0350b-ac81-4799-8aa6-c1205868586f.png)

2. I also get these logs every 10 seconds or even more often:

2023-02-17T11:58:41Z    INFO    Reconciling ScaledObject        {"controller": "scaledobject", "controllerGroup": "keda.sh", "controllerKind": "ScaledObject", "ScaledObject": {"name":"xxx","namespace":"xxx"}, "namespace": "xxx", "name": "data-ingest-iot", "reconcileID": "e272cdaa-8474-421e-9ca4-9fae7bda42f4"}

Which is probably strange since the default check period is every 30 seconds. I have a service bus trigger on another ScaledObject and neither of these things happen (no logs for reconciling, no "off" values either)

3. It seems that the value of the metric is capped at unprocessedEventThreshold * maxReplicaCount as very often I get value 40000 which is quite strange. So I think the value should not be capped. The value itself seems to be off as well, somewhat like in the point 1. but I cannot see it since it is capped.

### Expected Behavior

1. There should not be any values that are "off"
2. There probably should not be reconciling every 10 seconds in the log (not sure about that)
3. There should not be any "cap" on the metric value

### Actual Behavior

1. Strange values appear periodically
2. Logs indicate that probably something is wrong
3. The value of the metric is capped

### Steps to Reproduce the Problem

1. Create an eventhub with some messages coming through
2. Register trigger with the details posted above
3. Run for some time and observe this behaviour

P.S. The negative value does not seem to happen in every environment where I run k8s.

### Logs from KEDA operator

```
example
```

### KEDA Version

2.9.3

### Kubernetes Version

1.24

### Platform

Microsoft Azure

### Scaler Details

EventHubs

**Comment by JorTurFer** [comment]
Does this behavior still happen in v2.10?

**Comment by ilya-scale** [comment]
I have to upgrade to verify, let me check

**Comment by ilya-scale** [comment]
I am running in a development environment the upgraded version (2.10.1) and it actually does not seem to happen anymore. I need some more time to verify that this is indeed the case and upgrade production before I can tell for sure. 

Was there something that was fixed in 2.10 to make that happen?

**Comment by JorTurFer** [comment]
If I'm not wrong (and I can be), we upgraded the dep `azure-amqp-common-go` from v3 to v4

**Comment by ilya-scale** [comment]
It seems it did not help after all:

<img width="930" alt="image" src="https://user-images.githubusercontent.com/105273784/234203721-2d6c807e-533f-4d19-9d74-46b01976c352.png">

The 0 on the chart is not the 0, it is some real value (around 200), but as you see sometimes it goes to a completely weird value

**Comment by JorTurFer** [comment]
If you check the values using prometheus directly, do you see the same? I have had some renderization issues with grafana some times

**Comment by ilya-scale** [comment]
I am not using a dashboard, but a raw Metric for explore:

<img width="1845" alt="image" src="https://user-images.githubusercontent.com/105273784/234209600-0411ea5b-dbf2-40f7-b4b6-0b385a473430.png">

Would that be enough, or you think it can still render wrong?

**Comment by JorTurFer** [comment]
I still think that it can be a render error, I have faced with them sometimes in grafana :/ 
The problem is that value doesn't make sense :( maybe you could try with different time windows when you see the error to ensure that the negative peaks are at the same time. I mean, if you have the last 3 hours and you see an error 20 minutes ago, try again with the last hour and check if the error is at the same time (if the peak is at the same time with 2 different visualizations, looks like  metric problem)

**Comment by ilya-scale** [comment]
I have some issues with persistence, so I cannot go back in time to check unfortunately, but let me wait until it pops up again, and I can check raw prometheus data from prometheus pod to double check

**Comment by ilya-scale** [comment]
Here is the result from raw prometheus data (grafana is not involved):

<img width="1892" alt="image" src="https://user-images.githubusercontent.com/105273784/234260552-7938053e-6234-4919-afb5-c92df511056a.png">

So I guess that proves the issue is with the raw data?

**Comment by JorTurFer** [comment]
So, definitively KEDA is who is exposing that value :/
Do you have any idea about why this could be happening @v-shenoy @tomkerkhove ?

**Comment by ilya-scale** [comment]
I can also add that my setup is 2 - 8  replicas, since I have 8 partitions in a hub. Usually it runs with 2 pods, each listening to 4 partitions each.

**Comment by ilya-scale** [comment]
I suppose the issue was not really fixed, just closed?

**Comment by JorTurFer** [comment]
sure!
One question, how often does it happen? we have recently released a product that uses EventHub, it's just to check it looking for this weird values

**Comment by ilya-scale** [comment]
I have now turned the whole setup off since it is not possible to use with this issue, but as far as I remember it happened relatively often, more than once a day.

**Comment by JorTurFer** [comment]
I'll check it tomorrow morning

**Comment by JorTurFer** [comment]
In KEDA v2.10.1 still happens:
![image](https://github.com/kedacore/keda/assets/36899226/2f7fff89-318a-498b-8ddf-6a4b1f6a9102)

Do you have any idea about the root cause @tomkerkhove @v-shenoy . I guess that it's related with the circular buffer of the EventHub, but I'm not sure

**Comment by ilya-scale** [comment]
I think this is still an active issue that should not be closed

**Comment by JorTurFer** [comment]
@tomkerkhove , do you think that we could get a revision from anyone with knowledge about the eventhub SDK?

**Comment by tomkerkhove** [comment]
Can you summarize the issue please?

**Comment by JorTurFer** [comment]
The OP is using evenhub scaler and they monitor the exposed value during the time through KEDA's prometheus metrics. The problem is that periodically, KEDA is reporting a weird value:
![image](https://user-images.githubusercontent.com/105273784/219648250-e2d0350b-ac81-4799-8aa6-c1205868586f.png)

I have seen the same weird behavior in our own clusters (SCRM) and I have the suspicious that it's because we are doing something wrong calculating the lag.
